### PR TITLE
Add formatting guide

### PR DESCRIPTION
**Pull request by msujew**
Explains how an adopter would customize their own formatter for their language.

### CODE REVIEW

File: hugo/content/guides/formatting.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
+---
+title: "Formatting"
+weight: 300
+---
+
+Langium's formatting API allows to easily create formatters for your language.
+We start building a custom formatter for our language by creating a new class that inherits from `AbstractFormatter`.
+
+```ts
```
Comment (pluralia) [inline hugo/content/guides/formatting.md:9]: Could you specify where to put the `CustomModelFormatter` class? Now it looks like it has to be in one file with `CustomModelModule`.

File: hugo/content/guides/formatting.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+import { AbstractFormatter, AstNode, Formatting } from 'langium';
+
+export class CustomModelFormatter extends AbstractFormatter {
+    protected format(node: AstNode): void {
+        // This method is called for every AstNode in a document
+    }
+}
+...
+// Bind the class in your module
+export const CustomModelModule: Module<CustomModelServices, PartialLangiumServices> = {
```
Comment (pluralia) [inline hugo/content/guides/formatting.md:19]: Coma in this line is highlighted:
Screenshot 2022-11-23 at 15 34 46

File: hugo/content/guides/formatting.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    // Prepend all these nodes with no indent
+    nodes.prepend(Formatting.noIndent());
+}
+```
+
+Our other elements, namely `Entity` and `PackageDeclaration`, can be arbitrarily deeply nested, so using `noIndent` is out of the question for them.
+Instead we will use `indent` on everything between the `{` and `}` tokens. The formatter internally keeps track of the current indentation level:
+
+```ts
+if (ast.isEntity(node) || isPackageDeclaration(node)) {
```
Comment (pluralia) [inline hugo/content/guides/formatting.md:57]: This code differs with what we have for the domain model in the repo [here](https://github.com/langium/langium/blob/main/examples/domainmodel/src/language-server/domain-model-formatter.ts). Maybe update repo's version?

File: hugo/content/guides/formatting.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    formatter.interior(bracesOpen, bracesClose).prepend  (Formatting.indent());
+    // Also move the newline to a closing brace
+    bracesClose.prepend(Formatting.newLine());
+    // Surround the name property of the element
+    // With one space to each side
+    formatter.property("name").surround(Formatting.oneSpace());
+}
+```
+
+Note that most predefined `Formatting` methods accept additional arguments which make the resulting formatting more lenient.
```
Comment (pluralia) [inline hugo/content/guides/formatting.md:73]: Do you mind to tell more about these methods? Yes, it's a guide and not a documentation, but maybe it's a nice to see a couple more examples of API usage.

File: hugo/content/guides/formatting.md (hunk 1 above)
Comment (pluralia) [reply hugo/content/guides/formatting.md:9]: I'd propose don't use `CustomModelFormatter` -- you do all explanations using the domain model, so, maybe it's nice to tell about `DomainModelFormatter` from the beggning.

Comment (pluralia) [reply hugo/content/guides/formatting.md:9]: Also, looks like the `CustomModelFormatter` and `CustomModelModule` has to be called `CustomFormatter` and `CustomModule`.

File: hugo/content/guides/formatting.md (hunk 2 above)
Comment (montymxb) [reply hugo/content/guides/formatting.md:19]: This is an existing issue with the syntax highlighting. I've seen it in other changes as well. We should address it separately.

File: hugo/content/guides/formatting.md (hunk 3 above)
Comment (msujew) [reply hugo/content/guides/formatting.md:57]: We should do that, but it's unrelated to this PR.

**Review by pluralia — APPROVED** [review]
Approved ✅

**Pull request merged**